genome_grist/genbank_genomes.py

```diff
@@ -30,16 +30,17 @@ def url_for_accession(accession):
 
     full_name = None
     for line in all_names.splitlines():
-        name = line.split()[-1]
-        db_, acc_, *_ = name.split("_")
-        if db_ == db and acc_.startswith(acc):
-            full_name = name
-            break
+        if line.startswith(f'<a href='):
+            name=line.split('"')[1][:-1]
+            db_, acc_, *_ = name.split("_")
+            if db_ == db and acc_.startswith(acc):
+                full_name = name
+                break
 
     if full_name is None:
         return None
     else:
-        url = "ftp" + url[3:]
+        url = "htt" + url[3:]
```

Comment:
shouldn't this be
`url = "https" + url[3:]`?
I'm not sure if this line is tested, by `htt://ftp.ncbi.nlm.nih.gov/genomes/all/` is not going to be a valid URL...

Reply:
@luizirber the `url[3:]` cuts off the first 3 letters of the url :) so the `'htt'` is added to it.

for example:
```
>>> url_new = 'htt' + url[3:]
>>> url_new
'https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/009/494/275'
```

Reply:
But where is the `ps` coming from? (because the original url is `ftp://...`, so `url[3:]` is `://...`)

Reply:
Hm, when I tested the url parsing, the address changed to `url=https://` from the change in line 21, which then got cut off to `ps://` in line 43. I'm not sure that the slice is needed actually, but just in case it is, i stuck to fixing the prefix so the url would be complete.

Reply:
The tests do test this code, and I re-ran `make test` manually --

```
sample: HSMA33MX
outdir: outputs.test.HSMA33MX
sample: HSMA33MX
outdir: outputs.test.HSMA33MX
Job counts:
        count   jobs
        1       download_matching_genomes_one_by_one
        1
Job counts:
        count   jobs
        1       download_matching_genomes_one_by_one
        1
downloading genome for acc GCA_009494275.1/Prevotella copri from NCBI...
downloading genome for acc GCA_001881345.1/Escherichia coli from NCBI...
...wrote 1109089 bytes to genbank_genomes/GCA_009494275.1_genomic.fna.gz
...wrote 1581110 bytes to genbank_genomes/GCA_001881345.1_genomic.fna.gz
```
so it seems to work.

I did look at this when I reviewed it, and matched marisa's latest comment. It's definitely nonintuitive code though so next time I go through that section of code I'll take another look at it ;)